python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -1079,6 +1153,134 @@ def _get_task_parent_entity_type(self, task_id):
+        all_actions = dict()
+        config_names = []
+
+        for config_name, config_data in config_data.iteritems():
+            config_path, config_entity = config_data
+            commands = []
+            get_actions_data = project_actions[config_path]["shotgun_get_actions"]
+            env_file_name = "shotgun_%s.yml" % entity_type.lower()
+            raw_actions_data = get_actions_data[env_file_name]
+
+            if raw_actions_data["retcode"] != 0:
+                logger.error("A shotgun_get_actions call did not succeed: %s" % raw_actions_data)
+                continue
+
+            config_names.append(config_name)
+
+            for line in raw_actions_data["out"].split("\n"):
+                action = line.split("$")
+
+                if action[2] == "":
+                    deny_permissions = []
+                else:
+                    deny_permissions = action[2].split(",")
+
+                multi_select = action[3] == "True"
+
+                commands.append(
+                    dict(
+                        name=action[0],
+                        title=action[1],
+                        deny_permissions=deny_permissions,
+                        supports_multiple_selection=multi_select,
+                        app_name=None, # Not used here.
+                        group=None, # Not used here.
+                        group_default=None, # Not used here.
+                        engine_name=None, # Not used here.
+                    )
+                )
+
+            # We don't need or want the descriptor object to be sent to the client.
+            # Since we're done with this config for this invokation, we can just
+            # delete it from the entity dict.
+            del config_entity["descriptor"]
+
+            # And since we know this set of actions came from this legacy path, we
+            # can go ahead and include some extra data in the config dict that we
+            # can key off of when this action is called from the client.
+            config_entity["_legacy_config_root"] = config_path
+
+            all_actions[config_name] = dict(
+                actions=commands,
+                config=config_entity,
+            )
+
+        self.host.reply(
+            dict(
+                err="",
+                retcode=constants.SUCCESSFUL_LOOKUP,
+                actions=all_actions,
+                pcs=config_names,
+            ),
+        )
+
+    def _legacy_sanitize_output(self, out):
+        """
+        Sanitizes HTML output coming from the Shotgun engine by way of the
+        tank command. This method filters out any lines of output not wrapped
+        in span tags, and replaces HTML bold tags with Slack-style markdown
+        bold syntax.
+
+        :param str out: The raw output string to sanitize.
+
+        :returns: The sanitized output.
+        :rtype: str
+        """
+        sanitized = []
```

Comment:
`out.replace("<span>", "").replace("</span>", "")`?

Markdown supports html tags, so `<b>` is perfectly valid markdown, so no need to sanitize.